cvmfs/object_fetcher.h

```diff
@@ -195,8 +210,10 @@ class LocalObjectFetcher :
     file_path->clear();
 
     const std::string source = BuildPath(object_hash, hash_suffix);
-    const std::string dest   = temporary_directory_ + "/" +
-                               object_hash.ToString();
+    const std::string dest   = CreateTempPath(temporary_directory_ + "/" +
```

Comment:
Would be more secure to use CreateTempFile and to use the Decompress function working on files.  That also saves one pair of open/close functions.